scipy/signal/tests/test_splines.py

```diff
@@ -203,10 +203,13 @@ def test_symiir1_values(self, dtype, xp):
         atol = {xp.float64: 1e-15, xp.float32: 1e-7}[dtype]
         xp_assert_close(res, exp_res, atol=atol)
 
-        s = s + 1j*s
+        I1 = xp.asarray(
+            1 + 1j, dtype=xp.complex64 if s.dtype == xp.float32 else xp.complex128
+        )
```

Comment:
this looks slightly cleaner
```suggestion
        I1 = xp.asarray(
            1 + 1j, dtype=xp.result_type(s, xp.complex64)
        )
```
similar below

Reply:
Shrug. Feel free to push any cleanup you think is worth it.